src/BuildConfig.hpp

```diff
@@ -47,10 +47,16 @@ struct BuildConfig {
 
 private:
   std::string packageName;
+  std::string libName;
   fs::path buildOutPath;
   fs::path unittestOutPath;
   bool isDebug;
 
+  // if we are building an executable
+  bool executable{ false };
```

Comment:
Let's `binary` instead of `executable` as the term has already been used in the codebase. I think the comment for this variable is no longer needed, but I'd love to hear your thoughts.

```suggestion
  bool hasBinTarget{ false };
```

Reply:
Apologies for the delay, it was Thanksgiving here! Anyway, I think `executable` is much clearer than `binary` in this case. `Binary` is a very generic word but I do see that it's been used a lot elsewhere. I'm of the opinion that it should probably be changed elsewhere in the code too. But that might not be appropriate for this PR. I will use `binary`/`bin` for this PR to match the code and possibly change it later.